locallib.php

```diff
@@ -274,6 +274,14 @@ function equella_build_integration_url($args, $appendtoken = true) {
     return new moodle_url($CFG->equella_url, $equrlparams);
 }
 
+//Based on w3c standard, line breaks, as in multi-line text field values, are represented as CR LF pairs, i.e. `%0D%0A'
+//which represents "\r\n".
+//In order to align newline characters between serverside and client side, and make sure the signature is matched,
+//we should convert all "\n" to "\r\n".
+function convert_newline_characters($content) {
+    return str_replace("\n", "\r\n", $content);
+}
```

Comment:
Thanks, @ChristianMurphy. I am not able to insert CRLF in the DB, all I get is '\n', I guess it is related to OS? But I think your concern makes sense, it would be better to use preg in case someone harvests an item that includes CRLF.